hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/JoinHashPhysicalRel.java

```diff
@@ -58,9 +58,30 @@ public Join copy(
         return new JoinHashPhysicalRel(getCluster(), traitSet, left, right, conditionExpr, joinType);
     }
 
+    /**
+     * Cost calculation of Hash Join relation. It does not rely on children cost.
+     * <p>
+     * Hash Join algorithm is more advanced join algorithm, where it builds a hash table for left row set, and then
+     * compare each row from the right side
+     * Speaking of cost estimation, we are accounting the next properties:
+     * - produced row count is estimated as L * R, with assumption that the JOIN predicate has maximum selectivity.
+     * - processed row count is estimated as L + R, because we traverse both sides once per JOIN.
+     * - CPU cost estimation is a processed row multiplied by cost to build a hash table, and left and right rows comparison.
+     * <p>
+     * The perfect assumption also must include memory (what is important in case of hash table) and IO cost estimation,
+     * as well as a selectivity for a right row set.
```

Comment:
```suggestion
     * Hash Join algorithm is a more advanced join algorithm, which builds a hash table for the left
     * row set, and then compare each row from the right side. Cost estimation is the following: <ol>
     * <li> Produced row count is L * R assuming the join selectivity is 1.
     * <li> Processed row count is L + R because we traverse both sides once per JOIN.
     * <li> CPU is L * (hash table build cost) + R * (row comparison cost). </ol>
     * <p>
     * A perfect estimation must also include memory (occupied by the hash table) and IO costs,
     * as well as a selectivity for the right row set.
```